### PR TITLE
rgw_sal_motr:[CORTX-31962] Fix delete-marker's issue in case of suspended bucket.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
Problem:
In suspended bucket delete-marker's with new version-id is generated
instead of null version-id.

Solution:
If bucket is suspended then assign "null" value to delete-marker's
version-id.

Signed-off-by: Shriya Deshmukh <shriya.deshmukh@seagate.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by sachitanands — APPROVED** [review]
looks good

**Review by priyanka25081999 — APPROVED** [review]

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -2012,38 +2012,41 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
...
-        if (rc < 0) {
-          ldpp_dout(dpp, 0) << "Failed to delete the object from Motr. key- "<< delete_key << dendl;
-          return rc;
+          if (rc < 0) {
+            ldpp_dout(dpp, 0) << "Failed to delete the object from Motr. key- "<< delete_key << dendl;
+            return rc;
+          }
         }
-        // delete null version key and update null reference entry.
+        source->set_instance(del_marker_ver_id);
```
[the remark this answers is not in the document]
Comment (shriya-deshmukh) [reply src/rgw/rgw_sal_motr.cc:2032]: Yes here we are creating delete-marker and generating version-id for delete-marker, hence we are overwriting instance with del_marker_ver_id.

File: src/rgw/rgw_sal_motr.cc [hunk 2]
```diff
@@ -2012,38 +2012,41 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
...
+      if (rc < 0)
+        return rc;
       // creating a delete marker
       bufferlist del_mark_bl;
       rgw_bucket_dir_entry ent_del_marker;
       ent_del_marker.key.name = source->get_name();
-      ent_del_marker.key.instance = source->get_instance();
-      if (ent.key.instance == "null")
-        ent_del_marker.key.instance = "null";
+      ent_del_marker.key.instance = result.version_id;
```
[the remark this answers is not in the document]
Comment (cdeshmukh) [reply src/rgw/rgw_sal_motr.cc:2049]: so in versioning enabled or suspended case, this result.version_id will have expected version id set appropriate for both cases ?

File: src/rgw/rgw_sal_motr.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (shriya-deshmukh) [reply src/rgw/rgw_sal_motr.cc:2032]: Yes, this 'set_instance()' will be executed in both cases. this set_instance() method is used to form a key name (obj[delete-marker-version-id]) in any versioning case.
When versioning is suspended we need to show version-id as "null" for the delete-marker. Hence we need to assign the instance value(ent.key.instance) as null (line - 2049).

**Review by cdeshmukh — APPROVED** [review]

**Comment by cdeshmukh** [comment]
@shriya-deshmukh can you update build & deployment with latest code once, so we can merge this

**Comment by shriya-deshmukh** [comment]
Build and deployment  test result - https://jts.seagate.com/browse/CORTX-31962?focusedCommentId=2382465&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2382465

**Pull request merged**